File: reference/two-pointer/80-Remove-Duplicates-from-Sorted-Array-II.go
func removeDuplicates(nums []int) int {
    if len(nums) < 2 {
        return len(nums)
    }

    found := 2
    for i := 2; i < len(nums); i++ {
        if nums[i] != nums[found - 2] {
            nums[i], nums[found] = nums[found], nums[i]
            found++
        }
    }
    return found
}

File: reference/two-pointer/80-Remove-Duplicates-from-Sorted-Array-II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) <= 2:
            return len(nums)

        found = 2
        for i in range(2, len(nums)):
            if nums[i] != nums[found - 2]:
                nums[i], nums[found] = nums[found], nums[i]
                found += 1
        return found